src/os/shared.rs

```diff
@@ -32,23 +33,17 @@ impl Iterator for KeyboardEvents {
 
 fn get_datalink_channel(
     interface: &NetworkInterface,
-) -> Result<Box<dyn DataLinkReceiver>, failure::Error> {
+) -> Result<Box<dyn DataLinkReceiver>, std::io::Error> {
     let mut config = Config::default();
     config.read_timeout = Some(time::Duration::new(1, 0));
 
     match datalink::channel(interface, config) {
         Ok(Ethernet(_tx, rx)) => Ok(rx),
-        Ok(_) => failure::bail!("Unknown interface type"),
-        Err(e) => {
-            match e.kind() {
-                std::io::ErrorKind::PermissionDenied => failure::bail!("Failed to listen on network interface due to permission error. Try running with sudo"),
-                _ => failure::bail!(
-                    "Failed to listen on network interface {}: {}",
-                    interface.name,
-                    e
-                ),
-            }
-        }
+        Ok(_) => Err(std::io::Error::new(
```

Comment:
TBH I'm not sure if we should be using `std::io::Error` like this, from my point of view its use is reserved to the standard library. We should probably wrap `std::io::Error` in our own type, an enum with two variants, the other variant representing that we don't support that kind of interface. Ideally we would also say what kind of interface we don't support.

That being said, I think we can leave it like this for now and fix it in a followup task if we think it's worth it.

Reply:
I just realized I answered this in the other thread. sorry - it's been a day :)